apps/prairielearn/src/ee/lib/ai-grading.ts

```diff
@@ -44,34 +46,76 @@ const GradedExampleSchema = z.object({
   submission_text: z.string(),
   score_perc: z.number(),
   instance_question_id: z.string(),
+  manual_rubric_grading_id: z.string().nullable(),
 });
 type GradedExample = z.infer<typeof GradedExampleSchema>;
 
-async function generateGPTPromptWithExamples({
+const AppliedRubricItemSchema = z.object({
+  rubric_item_id: IdSchema,
+  score: z.coerce.number().nullish(),
+});
+type AppliedRubricItem = z.infer<typeof AppliedRubricItemSchema>;
+function parseRubricItems({
+  rubric_items,
+  gpt_rubric_items,
+}: {
+  rubric_items: RubricItem[];
+  gpt_rubric_items: number[];
+}): AppliedRubricItem[] {
+  const output: AppliedRubricItem[] = [];
+  for (const rubric_item_number of gpt_rubric_items) {
+    output.push(
+      AppliedRubricItemSchema.parse({ rubric_item_id: rubric_items[rubric_item_number].id }),
+    );
+  }
+  return output;
+}
+
+async function generateGPTPrompt({
   question_prompt,
   student_answer,
   example_submissions,
+  rubric_items,
 }: {
   question_prompt: string;
   student_answer: string;
   example_submissions: GradedExample[];
-}): Promise<
-  {
+  rubric_items: RubricItem[];
+}): Promise<{
+  messages: {
     role: 'system' | 'user';
     content: string;
-  }[]
-> {
+  }[];
+  warning: string;
+}> {
   const messages: {
     role: 'system' | 'user';
     content: string;
   }[] = [];
+  let warning = '';
 
   // Instructions for grading
-  messages.push({
-    role: 'system',
-    content:
-      'You are an instructor for a course, and you are grading assignments. You should always return the grade using a json object of 2 parameters: grade and feedback. The grade should be an integer between 0 and 100. 0 being the lowest and 100 being the highest, and the feedback should be why you give this grade, or how to improve the answer. You can say correct or leave blank when the grade is close to 100. I will provide some example answers and their corresponding grades.',
-  });
+  if (rubric_items.length) {
```

Comment:
```suggestion
  if (rubric_items.length > 0) {
```

I think this is a little more clear. Usually when you put an object straight into an if statement, it is mean to be a null/undefined check, but here we are actually doing a numberical check, right? Since rubric_items should never be null, but could be the empty array.